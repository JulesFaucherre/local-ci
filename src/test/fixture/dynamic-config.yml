version: 2.1

setup: true

jobs:
  set-up-config:
    executor: continuation/default
    steps:
      - checkout
      - run:
          name: Create config
          command: echo "Creating the dynamic config here"
          environment:
            BASE_REVISION: main
            MAPPING: |
              .*\.php$ run-php true
              .*\.js$ run-js true
              .*\.(php|js|json|yml)$ run-e2e true
            OUTPUT_PATH: /tmp/pipeline-parameters.json
      - run:
          name: Very simplified compiled circleci/continuation orb
          environment:
            CONFIG_PATH: .circleci/continue_config.yml
          command: $CIRCLE_CONTINUATION_KEY

workflows:
  setup:
    jobs:
      - set-up-config
