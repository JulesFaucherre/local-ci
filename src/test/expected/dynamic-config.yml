version: 2.1
setup: true
jobs:
  set-up-config:
    executor: continuation/default
    steps:
      - run:
          name: Ensure volume is writable
          command: |-
            if [ "$(ls -ld /tmp/local-ci | awk '{print $3}')" != "$(whoami)" ]
              then
              sudo chown $(whoami) /tmp/local-ci
            fi
      - checkout
      - run:
          name: Set more environment variables
          command: |-
            echo 'export CIRCLE_SHA1=$(git rev-parse HEAD)' >> $BASH_ENV
            echo 'export CIRCLE_BRANCH=$(git rev-parse --abbrev-ref HEAD)' >> $BASH_ENV
            echo 'export CIRCLE_PROJECT_REPONAME=$(basename $git remote get-url origin))' >> $BASH_ENV

      - run:
          name: Create config
          command: echo "Creating the dynamic config here"
          environment:
            BASE_REVISION: main
            MAPPING: |
              .*\.php$ run-php true
              .*\.js$ run-js true
              .*\.(php|js|json|yml)$ run-e2e true
            OUTPUT_PATH: /tmp/pipeline-parameters.json
      - run:
          name: Continue the pipeline
          command: |-
            if [ -f /tmp/local-ci/dynamic-config.yml ]
            then
              rm /tmp/local-ci/dynamic-config.yml
            fi
            cp .circleci/continue_config.yml /tmp/local-ci/dynamic-config.yml
            if [ -f /tmp/pipeline-parameters.json ]
            then
              cp /tmp/pipeline-parameters.json /tmp/local-ci/dynamic-config-parameters.json
            fi
workflows:
  setup:
    jobs:
      - set-up-config
